moon/controller.py

```diff
@@ -522,61 +522,83 @@ def get_angle_diff(self, destination, direction=1):
 
         return normalizeAnglePIPI(angle_diff)
 
-    def go_to_location(self, pos, desired_speed, offset=0.0, full_turn=False, timeout=None, with_stop=True, tolerance=1.0):
+    def get_avoidance_turn(self):
+        turn = None
+
+        # if avoidance expired, reset turn
+        if self.avoidance_start is not None and self.sim_time - self.avoidance_start >= timedelta(milliseconds=AVOIDANCE_DURATION):
+            self.avoidance_turn = None
+
+        # finish existing avoidance
+        if self.avoidance_start is not None:
+            if self.sim_time - self.avoidance_start < timedelta(milliseconds=AVOIDANCE_DURATION):
+                if self.sim_time - self.avoidance_start < timedelta(milliseconds=AVOIDANCE_TURN_DURATION):
+                    if self.debug:
+                        print(self.sim_time, self.robot_name, "Still turning")
+                    turn = self.avoidance_turn # keep turning 0.5s after obstacle no longer visible
+
+                elif self.sim_time - self.avoidance_start < timedelta(milliseconds=AVOIDANCE_DURATION):
+                    if self.debug:
+                        print(self.sim_time, self.robot_name, "Going straight")
+                    turn = GO_STRAIGHT # go straight additional 1s after no longer turning
+            else:
+                self.avoidance_turn = None
+
+        if self.scan_distance_to_obstacle < 4000:
+            # optionally do not avoid obstacles when real close to destination - if destination is a volatile, you know the obstacle is just a bump
+            self.avoidance_start = self.sim_time
+            if self.scan_avg_distance_left < 0.8 * self.scan_avg_distance_right:
+                self.avoidance_turn = -AVOID_RADIUS # steer to the right
+            elif self.scan_avg_distance_left * 0.8 > self.scan_avg_distance_right:
+                self.avoidance_turn = AVOID_RADIUS
+            else:
+                if self.avoidance_turn is None:
+                    self.avoidance_turn = AVOID_RADIUS if getrandbits(1) == 0 else -AVOID_RADIUS
+            turn = self.avoidance_turn
+            if (self.debug):
```

Comment:
you can still take into account `logging` with different verbosity levels

Reply:
yeah, crossed my mind as I was getting detailed logging about areas I wasn't interested in, after RC1 I think if there is time